### PR TITLE
Add RBAC support - issue 107

### PR DESCRIPTION
**Pull request by shanepeckham**
Changes include new directory for helm chart virtual-kubelet-0.1.1-rbac. A new entry has been added to the helm install command with switch rbac.install=<true|false> default false. For RBAC enabled clusters this switch needs to be set to true and this has been added to the readme in /providers/azure

Once approved this chart will need to be added to a tar file and merged to virtual-kubelet chart directory with version number 0.1.1

### CODE REVIEW

File: charts/virtual-kubelet-0.1.1-rbac/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{ if .Values.rbac.install }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "fullname" . }}-role-binding
+subjects:
+- kind: ServiceAccount
+  name: default
```
Comment (robbiezhang) [inline charts/virtual-kubelet-0.1.1-rbac/templates/clusterrolebinding.yaml]: shall we create a service account for VK? {{ template "fullname" }}-service-account?

File: charts/virtual-kubelet-0.1.1-rbac/values.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  nodeName: 
+  nodeTaint: 
+  nodeOsType: 
+  apiserverCert: 
+  apiserverKey: 
+
+## Install Default RBAC roles and bindings
+rbac:
+  install: false
+  serviceAccountName: default
```
Comment (robbiezhang) [inline charts/virtual-kubelet-0.1.1-rbac/values.yaml]: looks like not used, and I don't see a strong requirement to specify it when running the helm chart.

File: charts/virtual-kubelet-0.1.1-rbac/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+##############################################################################
+####  ERROR: You are missing required values in the values.yaml file.     ####
+##############################################################################
+
+This deployment will be incomplete until all the required fields in the values.yaml file have been provided.
+
+To update, run:
+
+    helm upgrade {{ .Release.Name }} \
+    --set env.azureClientId=<YOUR-AZURECLIENTID-HERE>,env.azureClientKey=<YOUR-AZURECLIENTKEY-HERE>,env.azureTenantId=<YOUR-AZURETENANTID-HERE>,env.azureSubscriptionId=<YOUR-AZURESUBSCRIPTIONID-HERE>,env.aciResourceGroup=<YOUR-ACIRESOURCEGROUP-HERE>,ev.aciOsType=<Linux|Windows>
```
Comment (robbiezhang) [inline charts/virtual-kubelet-0.1.1-rbac/templates/NOTES.txt]: please add the rbac values.

File: charts/virtual-kubelet-0.1.1-rbac/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+    spec:
```
Comment (robbiezhang) [inline charts/virtual-kubelet-0.1.1-rbac/templates/deployment.yaml]: add serviceAccountName explicitly.

Comment (shanepeckham) [reply charts/virtual-kubelet-0.1.1-rbac/templates/deployment.yaml]: Added serviceAccount from created name in values.yaml

File: charts/virtual-kubelet-0.1.1-rbac/templates/NOTES.txt (hunk 3 above)
Comment (shanepeckham) [reply charts/virtual-kubelet-0.1.1-rbac/templates/NOTES.txt]: Added rbac values

File: charts/virtual-kubelet-0.1.1-rbac/templates/clusterrolebinding.yaml (hunk 1 above)
Comment (shanepeckham) [reply charts/virtual-kubelet-0.1.1-rbac/templates/clusterrolebinding.yaml]: Implemented

File: charts/virtual-kubelet-0.1.1-rbac/values.yaml (hunk 2 above)
Comment (shanepeckham) [reply charts/virtual-kubelet-0.1.1-rbac/values.yaml]: Now referenced

**Comment by robbiezhang** [comment]
LGTM

**Pull request merged**